Updated Polish translation & fixed strings not being translated properly

eveninglatte opened the pull request:
Added some strings and fixed some old ones in the Polish translation.

While translating, I noticed three strings that are not getting translated properly on the Windows version:
1. "Restart Graphics" in the "Debug" window menu (`MainWindowMenu.cpp`):
![image](https://github.com/hrydgard/ppsspp/assets/87604676/2eba00bd-a39a-4ea1-a8e5-e5b8883fad93)

2. "ISO OK according to the Redump project" (along with the commented out "Game ID unknown - not in the Redump database") after triggering a CRC calculation for a game (`GameScreen.cpp`):
![image](https://github.com/hrydgard/ppsspp/assets/87604676/9e717daf-326d-4513-895d-c9e434f8cd70)

3. "Smart 2D texture filtering" in the Game settings -> Texture Filtering window menu:
![image](https://github.com/hrydgard/ppsspp/assets/87604676/1c4bdfe3-d01b-4ca7-af90-7806abd8ce75)

I was able to fix these for the Windows version (**I'm currently unable to test and possibly fix it for other platforms**) by:
1. Adding a `TranslateMenuItem()` method for `ID_DEBUG_RESTARTGRAPHICS`,
2. Changing strings passed to the `T()` function of the `I18NCategory` class,
3. Changing the "Smart 2D Texture Filtering" token to be capitalised on every letter in each location.

For the sake of consistency, I also capitalised every letter of "Smart 2D Texture Filtering" in the `[Graphics]` section and updated the `T()` call accordingly (though it does get translated correctly without these changes, as you can see in the background of the last picture).

eveninglatte committed:
Updated Polish translation

hrydgard commented:
Thank you for the fixes!

Would you mind changing the capitalization the other way though, to always lowercase except the first word? I'm (slowly) moving towards less excessive capitalization. Capitalizing every word used to be the most common style for settings and menus in Windows, but it's falling out of favor.

eveninglatte committed:
Fixed some strings not being translated properly
Also changed all occurences of "Smart 2D Texture Filtering" to be lowercase except for the first word

eveninglatte commented:
Sure thing, just pushed the changes.
